lib/rules/accessor-pairs.js

```diff
@@ -66,80 +66,84 @@ function isPropertyDescriptor(node) {
+module.exports = {
+    metadata: {
+        errors: {
+            getterMissing: "Getter is not present",
+            setterMissing: "Setter is not present"
+        },
+        description: "enforces getter/setter pairs in objects",
+        ruleType: "Best Practices",
+        recommended: false,
+        fixable: false
+    },
+    execute: function(context) {
+        /**
+        * Checks a object expression to see if it has setter and getter both present or none.
+        * @param {ASTNode} node The node to check.
+        * @returns {void}
+        * @private
+        */
+        function checkLonelySetGet(node) {
+            var isSetPresent = false;
+            var isGetPresent = false;
+            var isDescriptor = isPropertyDescriptor(node);
+
+            for (var i = 0, end = node.properties.length; i < end; i++) {
+                var property = node.properties[i];
+
+                var propToCheck = "";
+                if (property.kind === "init") {
+                    if (isDescriptor && !property.computed) {
+                        propToCheck = property.key.name;
+                    }
+                } else {
+                    propToCheck = property.kind;
                 }
-            } else {
-                propToCheck = property.kind;
-            }
 
-            switch (propToCheck) {
-                case "set":
-                    isSetPresent = true;
-                    break;
+                switch (propToCheck) {
+                    case "set":
+                        isSetPresent = true;
+                        break;
 
-                case "get":
-                    isGetPresent = true;
-                    break;
+                    case "get":
+                        isGetPresent = true;
+                        break;
 
-                default:
-                    // Do nothing
-            }
+                    default:
+                        // Do nothing
+                }
 
-            if (isSetPresent && isGetPresent) {
-                break;
+                if (isSetPresent && isGetPresent) {
+                    break;
+                }
             }
-        }
-
-        if (checkSetWithoutGet && isSetPresent && !isGetPresent) {
-            context.report(node, "Getter is not present");
-        } else if (checkGetWithoutSet && isGetPresent && !isSetPresent) {
-            context.report(node, "Setter is not present");
-        }
-    }
 
-    return {
-        "ObjectExpression": function(node) {
-            if (checkSetWithoutGet || checkGetWithoutSet) {
-                checkLonelySetGet(node);
+            if (context.options.setWithoutGet && isSetPresent && !isGetPresent) {
+                context.report(node, this.metadata.errors.getterMissing);
+            } else if (context.options.getWithoutSet && isGetPresent && !isSetPresent) {
+                context.report(node, this.metadata.errors.setterMissing);
```

Comment:
Still not sure about error messages. For most of our rules `this` would not be the rule definition but rather the object returned from the function. That makes it a tough proposition for automatic conversion.